src/solver/td3.ml

```diff
@@ -275,6 +281,21 @@ module Base =
       let destabilize_ref: (S.v -> unit) ref = ref (fun _ -> failwith "no destabilize yet") in
       let destabilize x = !destabilize_ref x in (* must be eta-expanded to use changed destabilize_ref *)
 
+      let format_wpoint x = Option.map_default (fun x -> Printf.sprintf "true (gas: %d)" x) "false" (HM.find_option wpoint_gas x) in
```

Comment:
If this is just used for tracing, then something like the following would be nicer:
```suggestion
      let pretty_wpoint () x =
        match HM.find_option wpoint_gas x with
        | None -> Pretty.text "false"
        | Some x -> Pretty.dprintf "true (gas: %d)" x
      in
```
(Don't commit this directly, I have not typechecked this.)

That can be used in tracing with `%a`. It avoids the lookup and string construction if tracing is enabled but something other is being traced. Otherwise unnecessary garbage is created.